### PR TITLE
Removes AWS in favor of Storage::cloud()

### PR DESCRIPTION
**Pull request by kedarv**
- Replaces existing AWS/s3 code in favor of Laravel Storage class in User model and SponsorsController
- Removes AWS dependency, and adds tests for the User Model.

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -279,6 +280,7 @@ public function getJWTIdentifier()
     public function getJWTCustomClaims()
     {
         $roles = $this->roles()->get()->pluck('name');
+
         return ['exp' => strtotime('+1 year'), 'roles'=>$roles, 'user_id'=>$this->id];
```
Comment (nickysemenza) [inline app/Models/User.php]: is this more readable as:
```
return [
    'exp'       => strtotime('+1 year'),
    'roles'     => $this->roles()->get()->pluck('name'),
    'user_id' => $this->id
];
```

File: composer.json [hunk 2]
```diff
@@ -6,7 +6,6 @@
     "type": "project",
     "require": {
         "php": "^7.1.3",
-        "aws/aws-sdk-php-laravel": "^3.1",
```
Comment (nickysemenza) [inline composer.json]: shouldn't the lockfile change too?

File: app/Models/User.php [hunk 3]
```diff
@@ -213,27 +213,28 @@ public function resumeURL($method = 'get')
...
-        $request = $s3->createPresignedRequest($cmd, '+7 days');
+
+        $client = Storage::cloud()->getDriver()->getAdapter()->getClient();
+        $command = $client->getCommand('PutObject', [
+            'Bucket' => getenv('AWS_BUCKET'),
+            'Key'    => $this->getResumeFilePath(),
+        ]);
+        $request = $client->createPresignedRequest($command, $expiry);
 
         return (string) $request->getUri();
```
Comment (nickysemenza) [inline app/Models/User.php]: can get rid of the string cast as storage layer already does that
https://github.com/illuminate/filesystem/blob/master/FilesystemAdapter.php#L534

Comment (kedarv) [reply app/Models/User.php]: We would still need this, since PutObject doesn't use the getAwsTemporaryUrl method.

File: app/Models/User.php (hunk 1 above)
Comment (kedarv) [reply app/Models/User.php]: ++

File: composer.json (hunk 2 above)
Comment (kedarv) [reply composer.json]: yep, good catch

**Pull request merged**